pkg/game/board.go

```diff
@@ -222,3 +222,138 @@ func (board *board) checkCompleted(
 	}
 	return scoreReport, nil
 }
+
+/*
+It analyzes road directed by roadSide parameter.
+It doesn't analyze starting tile.
+returns: road_finished, score, [meeples on road], loop, sideFinishedOn
+param roadSide: always indicates only one cardinal direction!
+*/
+func (board *board) CheckRoadInDirection(roadSide side.Side, startTile elements.PlacedTile) (bool, int, []elements.Meeple, bool, side.Side) {
+	var meeples = []elements.Meeple{}
+	var tile = startTile
+	var tileExists bool
+	var score = 0
+	var road *elements.PlacedFeature
+	var finished bool
+	// check finished on way
+	// do while loop
+	for {
+		tile, tileExists = board.GetTileAt(tile.Position.Add(elements.PositionFromSide(roadSide)))
+		roadSide = roadSide.ConnectedOpposite()
+		// check if tile exists or loop
+		if !tileExists || tile.Position == startTile.Position {
+			// tile does not exist
+			// finish
+			break
+		}
+
+		score++
+
+		// Get road feature
+		road = tile.GetPlacedFeatureAtSide(roadSide)
+
+		// check if there is meeple on the feature
+		if road.MeepleType != elements.NoneMeeple {
+			meeples = append(meeples, road.Meeple)
+		}
+
+		if road.Sides.GetCardinalDirectionsLength() == 1 {
+			// found the end of a road
+			break
+		}
+		// swap to other end of the road on the same tile
+		if road.Sides.GetNthCardinalDirection(0) == roadSide {
+			roadSide = road.Sides.GetNthCardinalDirection(1)
+		} else {
+			roadSide = road.Sides.GetNthCardinalDirection(0)
+		}
+
+	}
+	finished = tileExists && ((road.Sides.GetCardinalDirectionsLength() == 1) || (tile.Position == startTile.Position))
+
+	looped := (tile.Position == startTile.Position)
+	return finished, score, meeples, looped, roadSide
+}
+
+/*
+Calculates score for road.
+
+returns: ScoreReport, checked sides of the start tile (also including loop)
+*/
+func (board *board) ScoreRoadCompletion(tile elements.PlacedTile, road feature.Feature) (elements.ScoreReport, side.Side) {
+	var meeples = []elements.Meeple{}
+	var leftSide, rightSide side.Side
+	var score = 1
+	leftSide = road.Sides.GetNthCardinalDirection(0)
+	rightSide = road.Sides.GetNthCardinalDirection(1)
+	var roadFinished = true
+
+	var roadFinishedResult bool
+	var scoreResult int
+	var meeplesResult []elements.Meeple
+	var loopResult bool
+	var loopSide side.Side
+
+	// check meeples on start tile
+	var roadLeft = tile.GetPlacedFeatureAtSide(leftSide)
+	var roadRight = tile.GetPlacedFeatureAtSide(rightSide)
+	if roadLeft.MeepleType != elements.NoneMeeple || (roadRight != nil && roadRight.MeepleType != elements.NoneMeeple) {
+		if roadLeft.MeepleType != elements.NoneMeeple {
+			meeples = append(meeples, roadLeft.Meeple)
+		} else {
+			meeples = append(meeples, roadRight.Meeple)
+		}
+	}
```

Comment:
Please consider rewriting this to:
```
if roadLeft.MeepleType != elements.NoneMeeple {
	meeples = append(meeples, roadLeft.Meeple)
} else if roadRight != nil && roadRight.MeepleType != elements.NoneMeeple {
	meeples = append(meeples, roadRight.Meeple)
}
```
(shorter, doesn't have a big compound if condition, and doesn't check if roadLeft has a meeple twice)

Reply:
rewritten